Src/VimCore/RegisterMap.fs

```diff
@@ -100,9 +100,19 @@ type internal RegisterMap (_map : Map<RegisterName, Register>) =
     /// Updates the given register with the specified value.  This will also update 
     /// other registers based on the type of update that is being performed.  See 
     /// :help registers for the full details
-    member x.SetRegisterValue (reg : Register) regOperation (value : RegisterValue) = 
+    member x.SetRegisterValue (register : Option<Register>) regOperation (value : RegisterValue) (clipboardOptions : Vim.ClipboardOptions) = 

```

Comment:
Yeah,  I had a bad feeling doing this change, but i have no idea how can i do it better.

Do you mean something like that?
* the function SetRegisterValue takes a list of registers (or may be one register) to update, and updates only those/this registers without any extra logic
* some other function, which depend on configuration, parses user input, converts it to a list of registers for update and calls for each register the function SetRegisterValue.